### PR TITLE
Add node js 84.bot authentication msgraph

### PR DESCRIPTION
**Pull request by stevkan**

### CODE REVIEW

File: javascript_nodejs/84.bot-authentication-msgraph/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  cd javascript_nodejs/84.bot-authentication-msgraph
+  ```
+- Install modules
+  ```bash
+  npm install
+  ```
+- Update `basic-bot.bot` with required configuration settings
+  - App ID and Key for registered bots
+- Train LUIS to use the `greeting.lu` training set
+  ```bash
```
Comment (cleemullins) [inline javascript_nodejs/84.bot-authentication-msgraph/README.md]: This probably isn't right.

File: javascript_nodejs/84.bot-authentication-msgraph/README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+## Testing the bot using Bot Framework Emulator
+[Microsoft Bot Framework Emulator](https://github.com/microsoft/botframework-emulator) is a desktop application that allows bot developers to test and debug their bots on localhost or running remotely through a tunnel.
+
+- Install the Bot Framework Emulator from [here](https://aka.ms/botframework-emulator)
+
+### Connect to bot using Bot Framework Emulator v4
+- Launch Bot Framework Emulator
+- File -> Open Bot Configuration and navigate to `javascript_nodejs/84.bot-authentication-msgraph` folder
+- Select `authBot.bot` file
+
```
Comment (cleemullins) [inline javascript_nodejs/84.bot-authentication-msgraph/README.md]: Not authbot.Bot

File: javascript_nodejs/84.bot-authentication-msgraph/README.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+### Dependencies
+
+- **[Restify](http://restify.com)** Used to host the web service for the bot, and for making REST calls
+- **[dotenv](https://github.com/motdotla/dotenv)** Used to manage environmental variables
+
+### Project Structure
+
+The root `index.js` references the bot and starts a Restify server. The dialogs/mainDialog `index.js` loads the dialog type you selected when running the generator and adds it as the default dialog.
+
```
Comment (cleemullins) [inline javascript_nodejs/84.bot-authentication-msgraph/README.md]: grammar

File: javascript_nodejs/84.bot-authentication-msgraph/dialogs/mainDialog/index.js [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+// Microsoft Graph API to demonstrate making calls to authenticated services.
+class MainDialog {
+    /**
+         * Constructs the three pieces necessary by this bot to operate:
+         * 1. StatePropertyAccessor
+         * 2. DialogSet
+         * 3. ChoicePrompt
+         * 
+         * The only argument taken by this constructor is a ConversationState instance, although any BotState instance would suffice for this bot. `conversationState` is used to create a StatePropertyAccessor, which is needed to create a DialogSet. All botbuilder-dialogs `Prompts` need a DialogSet to operate.
+         * @param {ConversationState} conversationState
```
Comment (cleemullins) [inline javascript_nodejs/84.bot-authentication-msgraph/dialogs/mainDialog/index.js]: wrap it.

File: javascript_nodejs/84.bot-authentication-msgraph/dialogs/mainDialog/index.js [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+         */
+    constructor(conversationState) {
+        this.data = conversationState;
+        var that = this;
+
+        // DialogState property accessor. Used to keep persist DialogState when using DialogSet.
+        this.dialogState = conversationState.createProperty("dialogState");
+
+        // Create a DialogSet that will be contain the ChoicePrompt.
+        this.dialogs = new DialogSet(this.dialogState);
```
Comment (cleemullins) [inline javascript_nodejs/84.bot-authentication-msgraph/dialogs/mainDialog/index.js]: grammar. "will be contain"

File: javascript_nodejs/84.bot-authentication-msgraph/dialogs/mainDialog/index.js [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+
+        var graphData = {};
+        var photoResponse = {};
+        var emailResponse = {};
+        var helpMessage = `You can type "send <recipient_email>" to send an email, "recent" to view recent unread mail,` +
+            ` "me" to see information about your, or "help" to view the commands` +
```
Comment (cleemullins) [inline javascript_nodejs/84.bot-authentication-msgraph/dialogs/mainDialog/index.js]: const

File: javascript_nodejs/84.bot-authentication-msgraph/dialogs/mainDialog/index.js [hunk 7]
```diff
@@ -0,0 +1,423 @@
...
+                            "emailAddress": {
+                                "address": email
+                            }
+                        }
+                    ]
+                }
+            });
+
+            let optionsEmail = {
+                "uri": "https://graph.microsoft.com/v1.0/me/sendMail",
```
Comment (cleemullins) [inline javascript_nodejs/84.bot-authentication-msgraph/dialogs/mainDialog/index.js]: const

File: javascript_nodejs/84.bot-authentication-msgraph/README.md (hunk 3 above)
Comment (stevkan) [reply javascript_nodejs/84.bot-authentication-msgraph/README.md]: The index.js file is located in the 'dialogs/mainDialog' folder. I don't know why CodeFlow wasn't showing it.

**Comment by stevengum** [comment]
I did a pass this morning and cleaned up a bit, we'll iterate with any needed changes.

**Pull request merged**